content/client-lib-development-guide/features.textile

```diff
@@ -1242,6 +1255,30 @@ h4. TokenRequest
 * @(TE4)@ @ttl@ attribute represents time to live (expiry) of this token in milliseconds
 * @(TE6)@ @TokenRequest::fromJson@ static factory method that accepts either a @JsonObject@ or a string (which should be parsed as a JSON string), and returns a new @TokenRequest@. Statically typed languages (that expect the @authCallback@ to result in a typed @TokenRequest@ object, rather than, say, a hashmap) must implement this; others may at their discretion.
 
+h4. TokenRevocationRequest
+
+* @(TF1)@ @TokenRevocationRequest@ is a type containing the token revocation request details sent to the "REST revokeTokens endpoint":/rest-api/#revokeTokens
+* @(TF2)@ @TokenRevocationRequest#key@ attribute is a string representing key to revoke (appId.Key).
+* @(TF3)@ @TokenRevocationRequest#targets@ is an array of @TokenRevocationTarget@s to revoke. For request to be valid, it must contain at least one target. A test must exist to ensure there is at least one target in the request before it is sent.
+* @(TF4)@ @TokenRevocationRequest#issuedBefore@ is an optional timestamp (in milliseconds since the epoch) to revoke tokens issued before this time.
+* @(TF5)@ @TokenRevocationRequest#allowReauthMargin@ is an optional boolean that permits a token renewal cycle to take place without needing established connections to be dropped. Default for this value is false if not provided.
+
+h4. TokenRevocationTarget
+
+* @(TL1)@ @TokenRevocationTarget@ is a type representing a target for "REST revokeTokens endpoint":/rest-api/#revokeTokens.
+* @(TL2)@ @TokenRevocationTarget#key@ is a string that is the target key.This value cannot be null.
```

Comment:
The key again, in the target specifier? Why?

Reply:
This is the key as in target key like for example key for "clientId:client1@example.com"  is "clientId"

Reply:
I added some explanation on usage c4067a078a685bef9837339a736503c9cc5f7ff1. This is hopefully more clear now

Reply:
I'm really not keen on overloading `key` in this way. I think it's confusing. Can we not find a different word? Isn't it an array of specifiers `<type>:<target>`, so we could use `type` instead of `key` ?

Reply:
I agree `type` makes more sense in this context. The change is here ff5c856d4b60859cc07d42845c5ba31faad17f82